bot/exts/fun/off_topic_names.py

```diff
@@ -117,25 +152,133 @@ async def delete_command(self, ctx: Context, *, name: OffTopicName) -> None:
+
+        try:
+            await asyncio.wait_for(
+                rename_channel(),
+                3
+            )
+        except asyncio.TimeoutError:
+            # Channel rename endpoint rate limited. Cancel task and blacklist/de-activate name.
+            btn_yes = Button(label="Yes", style=ButtonStyle.success)
+            btn_no = Button(label="No", style=ButtonStyle.danger)
+
+            embed = Embed(
+                title=random.choice(NEGATIVE_REPLIES),
+                description=(
+
+                    "Re-naming the channel is being rate-limited. "
+                    "Would you like to schedule a channel re-name process within the current bot session ?"
+                ),
+                colour=Colour.blurple()
+            )
+
+            async def btn_call_back(schedule: bool, interaction: Interaction) -> None:
+
+                message = interaction.message
+
+                embed.description = (
+                    "Scheduled a channel re-name process within the current bot session."
+                    if schedule
+                    else
+                    "Channel not re-named due to rate limit. Please try again later."
+                )
+                await message.edit(embed=embed, view=None)
+
+                if schedule:
+                    await rename_channel()
+
+            btn_yes.callback = partial(btn_call_back, True)
+            btn_no.callback = partial(btn_call_back, False)
+
+            view = View()
+            view.add_item(btn_yes)
+            view.add_item(btn_no)
+
+            await ctx.message.reply(embed=embed, view=view)
+
+    @otname_group.group(name='list', aliases=('l',), invoke_without_command=True)
     @has_any_role(*MODERATION_ROLES)
     async def list_command(self, ctx: Context) -> None:
         """
         Lists all currently known off-topic channel names in a paginator.
 
         Restricted to Moderator and above to not spoil the surprise.
         """
-        result = await self.bot.api_client.get('bot/off-topic-channel-names')
-        lines = sorted(f"• {name}" for name in result)
-        embed = Embed(
-            title=f"Known off-topic names (`{len(result)}` total)",
-            colour=Colour.blue()
-        )
-        if result:
-            await LinePaginator.paginate(lines, ctx, embed, max_size=400, empty=False)
-        else:
-            embed.description = "Hmmm, seems like there's nothing here yet."
-            await ctx.send(embed=embed)
+        await self.active_otnames_command(ctx)
+
+    @list_command.command(name='active')
+    @has_any_role(*MODERATION_ROLES)
+    async def active_otnames_command(self, ctx: Context) -> None:
+        """List active off topic channel names."""
+        await self.list_ot_names(ctx, True)
+
+    @list_command.command(name='inactive')
+    @has_any_role(*MODERATION_ROLES)
+    async def inactive_otnames_command(self, ctx: Context) -> None:
+        """List inactive off topic channel names."""
```

Comment:
I think "inactive" more-so says that the off-topic channel name is somehow lazy. "deactivated" (or "disabled") on the other hand gets the point across that a moderator has actively seen the name and decided that it should not be seen by the rest of the community.